src/lib/redux/siteDataSlice.ts

```diff
@@ -41,8 +41,19 @@ const siteDataSlice = createSlice({
     setCurrSite(state, action) {
       state.currentSite = action.payload;
     },
+    addNewCustomer(state, action) {
```

Comment:
This nesting was based on the [article you linked](https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns#updating-nested-objects) -- it's definitely  not the cleanest way, but I decided not to dig into `createReducer` last night so if you think we should still switch to that and/or un-nest, lmk.

Reply:
IMO it'd be cleanest to unnest. That way we can have a `customerDataSlice` where it'd make sense to put `EMPTY_CUSTOMER` (might be better in utils but `EMPTY_SITE` is in `siteDataSlice`) and `addCustomer` makes more sense in that file. 

This might be a lot of refactor work, so maybe I could do it in a subsequent PR.

**NOTE**: We'd have to be careful, though when changing sites. We need to make sure that if a User is in charge of 2 sites, `Site A` and `Site B`, that the scenario where:
1.  I create a customer to `Site A`, 
2. Switch to `Site B` 
3. Switch back to `Site A`
that the customer I created in `Site A` is still visible (we can do this by persisting changes to `customers` into `currentSite` whenever a site change is made)